autokeras/text/text_supervised.py

```diff
@@ -174,15 +117,29 @@ def predict(self, x_test):
 
             logits = logits.detach().cpu().numpy()
             y_preds.extend(logits)
-        return self.inverse_transform_y(y_preds)
+
+        if self.task_type:
+            return self.inverse_transform_y(y_preds)
+        else:
+            return y_preds
 
     @property
     def metric(self):
```

Comment:
This if else is now written in a procedure-oriented style.
For object-oriented programming, this function should be an abstract function.
The different subclasses should override it to return different values.
The same for the loss().